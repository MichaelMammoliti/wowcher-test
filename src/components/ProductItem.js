import React from 'react';

import { formatNumber } from '../utilities';

const ProductItem = ({ productData }) => (
  productData.map((productItem) => (
    <tr key={productItem.name}>
      <td>{productItem.name}</td>
      <td>{formatNumber(productItem.unitPrice * productItem.sold)}</td>
    </tr>
  )
));

export default ProductItem;
