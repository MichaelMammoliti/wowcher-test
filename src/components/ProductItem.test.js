import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";
import "jest-enzyme";

import ProductItem from './ProductItem';

configure({
  adapter: new Adapter()
});

describe('ProductItem', () => {
  it('should render correctly', () => {
    const props = {
      productData: [
        {
          name: 'somethig',
          unitPrice: 10,
          sold: 10,
        }
      ]
    };

    const unit = shallow(<ProductItem {...props} />);

    expect(unit).toMatchSnapshot();
  });
})

export default ProductItem;
