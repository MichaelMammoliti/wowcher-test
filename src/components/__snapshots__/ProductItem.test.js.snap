// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductItem should render correctly 1`] = `
<tr
  key="somethig"
>
  <td>
    somethig
  </td>
  <td>
    100.00
  </td>
</tr>
`;
